be/src/olap/tablet.cpp

```diff
@@ -252,28 +155,6 @@ OLAPStatus Tablet::add_rowset(RowsetSharedPtr rowset, bool need_persist) {
     _rs_version_map[rowset->version()] = rowset;
     RETURN_NOT_OK(_rs_graph.add_version_to_graph(rowset->version()));
 
-    vector<RowsetSharedPtr> rowsets_to_delete;
-    // yiguolei: temp code, should remove the rowset contains by this rowset
-    // but it should be removed in multi path version
-    for (auto& it : _rs_version_map) {
-        if ((it.first.first >= rowset->start_version() && it.first.second < rowset->end_version())
-            || (it.first.first > rowset->start_version() && it.first.second <= rowset->end_version())) {
-            if (it.second == nullptr) {
-                LOG(FATAL) << "there exist a version "
-                           << " start_version=" << it.first.first
-                           << " end_version=" << it.first.second
-                           << " contains the input rs with version "
-                           << " start_version=" << rowset->start_version()
-                           << " end_version=" << rowset->end_version()
-                           << " but the related rs is null";
-                return OLAP_ERR_PUSH_ROWSET_NOT_FOUND;
-            } else {
-                rowsets_to_delete.push_back(it.second);
-            }
-        }
-    }
-    modify_rowsets(std::vector<RowsetSharedPtr>(), rowsets_to_delete);
-
     if (need_persist) {
         RowsetMetaPB rowset_meta_pb;
```

Comment:
Callers set need_persist to true will persist rowset meta to meta store.

Reply:
I will set need_persist in compaction